xds/internal/xdsclient/authority.go

```diff
@@ -117,14 +118,34 @@ func newAuthority(args authorityArgs) (*authority, error) {
 		ServerCfg:          *args.serverCfg,
 		UpdateHandler:      ret.handleResourceUpdate,
 		StreamErrorHandler: ret.newConnectionError,
+		OnSendHandler:      ret.transportAfterSendHandler,
 		Logger:             args.logger,
 	})
 	if err != nil {
 		return nil, fmt.Errorf("creating new transport to %q: %v", args.serverCfg, err)
 	}
 	ret.transport = tr
 	return ret, nil
+}
 
+// transportAfterSendHandler is called by the underlying transport when it sends a
+// resource req successfully. Timers are started for resources waiting for a response.
+func (a *authority) transportAfterSendHandler(u *transport.UpdateChannelInfo) {
+	a.resourcesMu.Lock()
+	defer a.resourcesMu.Unlock()
+	rType := a.resourceTypeGetter(u.URL)
+	rTypeMap := a.resources[rType]
+
+	for _, resourceName := range u.ResourceNames {
+		if state, ok := rTypeMap[resourceName]; ok {
+			if state.wState == watchStateStarted {
```

Comment:
Don't we have to handle the case where the state is `watchStateRespRequested` by restarting the timer?

Like we could have the following scenario:
- watch started for resource A
- request sent out for resource A
- response received for resource A
- watch started for resource B
- request sent for resource B
- stream breaks
  - `onConnectionError` should not be called here because we already received a message on the stream

At this point, let's say that the stream restarts and resends A & B, and we get this callback. We should reset the timer for resource B, correct?